docs/how-to/hip_runtime_api/error_handling.rst

```diff
@@ -0,0 +1,212 @@
+--------------------------------------------------------------------------------
+
+1. Check errors after each API call
+
+  Always check the return value of HIP API calls to catch errors early. For
+  example:
+
+  .. code-block:: cpp
+
+    hipError_t err = hipMalloc(&d_A, size);
+    if (err != hipSuccess) {
+        std::cerr << "hipMalloc failed: " << hipGetErrorString(err) << std::endl;
+        return -1;
+    }
+
+2. Use macros for error checking
+
+  Define macros to simplify error checking and reduce code duplication. For
+  example:
+
+  .. code-block:: cpp
+
+    #define HIP_CHECK(expression)                  \
+    {                                              \
+        const hipError_t status = expression;      \
+        if(status != hipSuccess){                  \
+            std::cerr << "HIP error "              \
+                      << status << ": "            \
+                      << hipGetErrorString(status) \
+                      << " at " << __FILE__ << ":" \
+                      << __LINE__ << std::endl;    \
+        }                                          \
+    }
+
+    // Allocate memory on the device with error check.
+    HIP_CHECK(hipMalloc(&device_pointer, N * sizeof(float);));
+
+3. Handle errors gracefully
+
+   Ensure the application can handle errors gracefully, such as by freeing
+   resources or providing meaningful error messages to the user.
+
+Complete example
+--------------------------------------------------------------------------------
+
+A complete example demonstrating error handling:
+
+.. code-block:: cpp
+
+  #include <hip/hip_runtime.h>
+  #include <iostream>
+
+  #define HIP_CHECK(expression)                  \
+  {                                              \
+      const hipError_t status = expression;      \
+      if(status != hipSuccess){                  \
+          std::cerr << "HIP error "              \
+                    << status << ": "            \
+                    << hipGetErrorString(status) \
+                    << " at " << __FILE__ << ":" \
+                    << __LINE__ << std::endl;    \
+      }                                          \
+  }
+
+  // Addition of two values.
+  __global__ void add(int *a, int *b, int *c) {
+      *c = *a + *b;
+  }
+
+  int main() {
+      int a, b, c;
+      int *d_a, *d_b, *d_c;
+
+      // Setup input values.
+      a = 1;
+      b = 2;
+
+      // Allocate device copies of a, b and c.
+      HIP_CHECK(hipMalloc(&d_a, sizeof(*d_a)));
+      HIP_CHECK(hipMalloc(&d_b, sizeof(*d_b)));
+      HIP_CHECK(hipMalloc(&d_c, sizeof(*d_c)));
+
+      // Copy input values to device.
+      HIP_CHECK(hipMemcpy(d_a, &a, sizeof(*d_a), hipMemcpyHostToDevice));
+      HIP_CHECK(hipMemcpy(d_b, &b, sizeof(*d_b), hipMemcpyHostToDevice));
+
+      // Launch add() kernel on GPU.
+      hipLaunchKernelGGL(add, dim3(1), dim3(1), 0, 0, d_a, d_b, d_c);
+      // Check the kernel launch
+      HIP_CHECK(hipPeekAtLastError());
```

Comment:
Unless there is a reason for not resetting the error, I think hipGetLastError should be used. This way, if there was some kind of error handling, the subsequent function might fail with the same error again.
```suggestion
      HIP_CHECK(hipGetLastError());
```